### DISCUSSION TITLE
How to govern Redis in Console project

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#397] [General]
### Description

Since the beginning of my experience with Mia Platform Console, I've been told not to install [Redis](https://redis.io/) as a micro-service inside my architecture, but rather to use the Redis instance running in the `default` namespace of the cluster.
I never asked questions, but now it seems to me this is not the right solution either. In most of our applications we use Redis to handle session data and the connection pod <-> Redis happens without any security measure: no credentials, no mutual authentication, nothing.

In my opinion, each architecture should have its dedicated instance of Redis, the same way we do with Mongo: each project a database, credentials, connection string and so on.

What's preventing us from setting up a Redis server for each architecture? (like shown [here](https://www.dragonflydb.io/guides/redis-kubernetes)) or to use a cloud SaaS like we do with Mongo Atlas?

**Comment by fredmaggiowski** [comment]
Hi, Mia-Platform PaaS used to provide a multitenant redis instance (generally not in the default namespace, but that depends on the cluster setup).

I feel like there is no definitive answer to this (a part from **not deploy it as a pod**).

It depends on how you wish to manage infrastructure. Depending on your needs, it may make sense to have a shared redis instance in a dedicated cluster, for a single company to be shared by multiple projets. 
In other cases (e.g. whenever you need to make sure data is safely isolated and/or require high throughput) a dedicated redis cluster might be a better solution. How to deploy it depends (again) on your infrastructure needs.

**Reply by theliuk** [reply]
I see,... I don't know. At the moment I feel there's no governance of the tool. Everybody use it, but without any discipline, so applications may access each other data and nobody would notice.

**Reply by fredmaggiowski** [reply]
@verdienansein @stefano-arre are you aware of this? 

I remember some policies has been introduced in regards of redis management within our PaaS proposal. I guess such issues could arise in the context of dedicated infrastructure but that of course depends on the policies in act in who is using such clusters.

**Reply by verdienansein** [reply]
I think that, in case of more complex architectures that needs Redis, a dedicated Redis should be provided with authentication and HA features.
As @fredmaggiowski already said, depending on your needs you should ask for a dedicated Redis cluster.
